compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -386,7 +469,7 @@ class SpaceEngine(implicit ctx: Context) extends SpaceLogic {
   /** Is `tp1` a subtype of `tp2`?  */
   def isSubType(tp1: Type, tp2: Type): Boolean = {
     // check SI-9657 and tests/patmat/gadt.scala
-    val res = erase(tp1) <:< erase(tp2)
+    val res = tp1 <:< erase(tp2)
```

Comment:
Could you add a comment explaining why it's correct to erase `tp2` but not `tp1` ?